### ISSUE TITLE
Gumbel Noise function

### ISSUE THREAD

**Issue by alvations** [awslabs/sockeye#1004]
Hi sockeye dev, if we want to add [gumbel noise function](https://github.com/pytorch/pytorch/issues/639)  for the pytorch backend sockeye, where should the function be included? Or is there already a similar function in sockeye?

```python
def gumbel_noise(input, epsilon=1e-8):
    return input.new_zeros(*input.size()).uniform_().add_(
        epsilon).log_().neg_().add_(epsilon).log_().neg_()
```

**Comment by mjdenkowski** [comment]
Is there a particular place you're interested in using Gumbel noise?

PyTorch includes support for sampling from [Gumbel distributions](https://pytorch.org/docs/stable/distributions.html#gumbel) and the [Gumbel softmax](https://pytorch.org/docs/stable/generated/torch.nn.functional.gumbel_softmax.html#torch-nn-functional-gumbel-softmax).

**Comment by mjdenkowski** [comment]
Closing for now.  Feel free to reopen if you have other questions.
